src/compiler/scanner.ts

```diff
@@ -605,12 +610,23 @@ module ts {
         }
 
         function isIdentifierStart(ch: number): boolean {
+            if (inJSXTag && ch === CharacterCodes.backslash) {

```

Comment:
I would avoid this.  Instead, just have the scanner track if it ran into a unicode escape while scanning an identiifer.  If so, we can either not allow the token to be used at hte parser level for a JSX tag.  Or, we can allow it, and we can appropriately emit hte cooked name instead of the raw name.